tests/test_trustdb.py

```diff
@@ -0,0 +1,563 @@
+    "go_reliability_data, expected_result",
+    [  # testcase1:No reports for the IP
+        (
+                "192.168.1.1",
+                [],
+                [],
+                [],
+                [],
+                [],
+        ),
+        # Testcase2:One report, reporter's IP matches target IP
+        (
+
+                "192.168.1.1",
+                [
+                    (
+                            "reporter_1",
+                            "ip",
+                            "192.168.1.1",
+                            0.5,
+                            0.8,
+                            1678886400,
+                    ),
+                ],
+                [
+                    ("192.168.1.1", "reporter_1", 1678886300),
+                ],
+                [],
+                [],
+                [],
+        ),
+        # Testcase3:One valid report
+        (
+                "192.168.1.1",
+                [
+                    (
+                            "reporter_1",
+                            "ip",
+                            "192.168.1.1",
+                            0.5,
+                            0.8,
+                            1678886400,
+                    ),
+                ],
+                [
+                    ("192.168.1.2", "reporter_1", 1678886300),
+                ],
+                [
+                    ("192.168.1.2", 0.6, 0.9, 1678886350),
+                ],
+                [(("reporter_1",), 0.7)],
+                [],
+        ),
+        # Testcase4:Multiple valid reports
+        (
+                "192.168.1.1",
+                [
+                    (
+                            "reporter_1",
+                            "ip",
+                            "192.168.1.1",
+                            0.5,
+                            0.8,
+                            1678886400,
+                    ),
+                    (
+                            "reporter_2",
+                            "ip",
+                            "192.168.1.1",
+                            0.3,
+                            0.6,
+                            1678886500,
+                    ),
+                ],
+                [
+                    ("192.168.1.2", "reporter_1", 1678886300),
+                    ("192.168.1.3", "reporter_2", 1678886450),
+                ],
+                [
+                    ("192.168.1.2", 0.6, 0.9, 1678886350),
+                    ("192.168.1.3", 0.2, 0.5, 1678886480),
+                ],
+                [(("reporter_1",), 0.7), (("reporter_2",), 0.4)],
+                [],
+        ),
+    ],
+)
+def test_get_opinion_on_ip(
+        mock_db,
+        ipaddress,
+        reports_data,
+        peer_ips_data,
+        slips_reputation_data,
+        go_reliability_data,
+        expected_result,
+):
+    mock_db.execute.side_effect = None
+    mock_db.fetchall.side_effect = [
```

Comment:
hey @Sekhar-Kumar-Dash i'm having trouble understanding this one, how come fetchall() is only called once in the original function, and it has 5 side effects in the test, and each time it returns a different data structure?
in my mind, if it's called once, it should have 1 side effect right?

Reply:
isn't it better to split this function into smaller ones?

Reply:
yes you are correct i will try to do it